src/GraphQL/Resolver/Element.php

```diff
@@ -81,5 +74,152 @@ public function resolveProperties($value = null, $args = [], $context, ResolveIn
         return $result;
     }
 
-}
+    /**
+     * @param null $value
+     * @param array $args
+     * @param $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return array
+     * @throws \Exception
+     */
+    public function resolveParent($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        $element = $this->getElementById($value['id']);
+        if ($element) {
+            $parent = $element->getParent();
+            if ($parent) {
+                return $this->extractSingleElement($parent, $args, $context, $resolveInfo);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param null $value
+     * @param array $args
+     * @param $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return array
+     * @throws \Exception
+     */
+    public function resolveChildren($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        $element = $this->getElementById($value['id']);
+        if ($element) {
+            $arguments = $this->composeArguments($args);
+            return $this->extractMultipleElements($element->getChildren(...$arguments), $args, $context, $resolveInfo);
+        }
+        return [];
+    }
+
+    /**
+     * @param null $value
+     * @param array $args
+     * @param $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return array
+     * @throws \Exception
+     */
+    public function resolveSiblings($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        $element = $this->getElementById($value['id']);
+        if ($element) {
+            $arguments = $this->composeArguments($args);
+            return $this->extractMultipleElements($element->getSiblings(...$arguments), $args, $context, $resolveInfo);
+        }
+        return [];
+    }
+
+    /**
+     * @param array $args
+     * @return array
+     */
+    protected function composeArguments($args = [])
+    {
+        $arguments = [];
+        if ($this->elementType === 'object') {
+            $arguments[] = isset($args['objectTypes']) ? $args['objectTypes'] : [AbstractObject::OBJECT_TYPE_OBJECT, AbstractObject::OBJECT_TYPE_FOLDER];
+        }
+        return $arguments;
+    }
+
+    /**
+     * @param $value
+     * @return AbstractElement|null
+     * @throws \Exception
+     */
+    protected function getElementById($id)
+    {
```

Comment:
we probably can replace this entire block with Pimcore\Model\Element\Service::getElementById()